hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -349,29 +441,49 @@ public X509Certificate queryCertificate(String query) {
   }
 
   /**
-   * Stores the Certificate  for this client. Don't use this api to add
-   * trusted certificates of other components.
+   * Stores the Certificate  for this client. Don't use this api to add trusted
+   * certificates of other s.
    *
-   * @param certificate - X509 Certificate
+   * @param pemEncodedCert - pem encoded X509 Certificate
+   * @param force - override any existing file
+   * @param isLocalIdentityCert - true if certificate belongs to the identity
+   * cert for this certificate client.
    * @throws CertificateException - on Error.
+   *
+   * Note: Certificate client can store certificates for other daemons as well.
+   * Local certificate refers to the certificate issued to this certificate
+   * client. This is stored along with public key and private key. Certificate
+   * of other daemons is stoed in sub dirs named after certificate serial id of
+   * certificate.
    */
   @Override
-  public void storeCertificate(X509Certificate certificate)
-      throws CertificateException {
+  public void storeCertificate(String pemEncodedCert, boolean force,
+      boolean isLocalIdentityCert) throws CertificateException {
     CertificateCodec certificateCodec = new CertificateCodec(securityConfig);
     try {
-      certificateCodec.writeCertificate(
-          new X509CertificateHolder(certificate.getEncoded()));
-    } catch (IOException | CertificateEncodingException e) {
+      Path basePath = securityConfig.getCertificateLocation();
+      String certName;
+      X509Certificate cert =
+          CertificateCodec.getX509Certificate(pemEncodedCert);
+      if (isLocalIdentityCert) {
+        certName = securityConfig.getCertificateFileName();
+      } else {
+        certName = String.format(CERT_FILE_NAME_FORMAT,
```

Comment:
You should use Paths.get() the path format with hard coded path separator won't work across platforms.

Reply:
CERT_FILE_NAME_FORMAT doesn't have any path separator in it. CertificateCodec class does make use of Paths.get when we pass basePath and file name.